discovery/azure/azure.go

```diff
@@ -291,6 +297,27 @@ func (d *Discovery) refresh() (tg *targetgroup.Group, err error) {
 							labels[azureLabelMachinePrivateIP] = model.LabelValue(*ip.Properties.PrivateIPAddress)
 							address := net.JoinHostPort(*ip.Properties.PrivateIPAddress, fmt.Sprintf("%d", d.port))
 							labels[model.AddressLabel] = model.LabelValue(address)
+							// Check if Public IP is also present
+							// Get the Public IP address information via separate call to the network provider.
```

Comment:
Azure has a quite low rate limit for API requests, so an additional request per instance let alone per network interface is problematic. Is there a way to do this without any additional requests?

Reply:
Unfortunately there is no other way as public IP is a separate resource in Azure RM API and only referenced by ID in the network interface.

Reply:
That's unfortunate, this isn't the first PR we've had to reject for this. Hopefully Azure will have higher API limits or more denormalised APIs in the future.

Reply:
Yes I understand. But I was thinking this network call will be triggered only for VMs which have a public IP. Most VMs should have only Private IP(internal). 
Thanks for quick review. Will close this.